eis_toolkit/deep_learning/mlp_function.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+from sklearn.neural_network import MLPClassifier
+
+from eis_toolkit.exceptions import InvalidArgumentTypeException, InvalidDatasetException
+from eis_toolkit.model_performance_estimation.model_performance_estimation import performance_model_estimation
+
+
+def train_evaluate_predict_with_mlp(
+    dataset: np.ndarray,
+    labels: np.ndarray,
+    cross_validation_type: str,
+    number_of_split: int,
+    is_class_probability: bool = False,
+    threshold_probability: float = None,
+    is_predict_full_map: bool = False,
+    solver: str = "adam",
+    alpha: float = 0.001,
+    hidden_layer_sizes: tuple[int, int] = (16, 2),
+    random_state=0,
+) -> np.ndarray:
+    """
+    Do the training - evaluation - predictions steps with MLP.
+
+    Parameters:
+        dataset: Features data.
+        labels: Labels data.
+        cross_validation_type: selected cross validation method.
+        number_of_split: number of split to divide the dataset.
+        is_class_probability: if True the code return probability, otherwise it return class.
+        is_predict_full_map: if True the function will predict the full dataset otherwise predict only the te4st fold.
+        threshold_probability: works only if is_class_probability is True, is thresholds of probability.
+        solver: this is what in keras is called optimizer.
+        alpha: floating point represent regularization.
+        hidden_layer_sizes: It represents the number of neurons in the ith hidden layer.
+        random_state: random state for repeatability of results.
+    Return:
+        a numpy array with prediction (class if is_class_probability is set to false otherwise it return probability).
+    Raises:
+        InvalidDatasetException: When the dataset is None.
+        InvalidArgumentTypeException when the function try to make probability and the threshold is None.
+    """
+
+    # I need two local vars
+    best_score = 0
+    best_handler_list = list()
+
+    if is_class_probability is not False and threshold_probability is None:
+        raise InvalidArgumentTypeException
```

Comment:
I'd use here the InvalidParameterValueException and include a comment in the exception